src/wasm-interpreter.h

```diff
@@ -2309,21 +2309,27 @@ template<typename GlobalManager, typename SubType> class ModuleInstanceBase {
   std::unordered_set<size_t> droppedSegments;
 
   void initializeTableContents() {
-    for (auto& segment : wasm.elementSegments) {
-      if (segment->table.isNull()) {
-        continue;
-      }
-
+    ModuleUtils::iterActiveElementSegments(wasm, [&](ElementSegment* segment) {
       Address offset =
         (uint32_t)InitializerExpressionRunner<GlobalManager>(globals, maxDepth)
           .visit(segment->offset)
           .getSingleValue()
           .geti32();
-      for (size_t i = 0; i != segment->data.size(); ++i) {
-        externalInterface->tableStore(
-          segment->table, offset + i, segment->data[i]);
+
+      for (Index i = 0; i < segment->data.size(); ++i) {
+        if (auto* get = segment->data[i]->dynCast<RefFunc>()) {
+          externalInterface->tableStore(
+            segment->table,
+            offset + i,
+            Literal::makeFunc(get->func, get->type));
+        } else if (segment->data[i]->is<RefNull>()) {
+          externalInterface->tableStore(
+            segment->table,
+            offset + i,
+            Literal::makeNull(segment->data[i]->type));
```

Comment:
It'd be much easier to do that once we have `table.init` instructions, in which case it could mirror `initializeMemoryContents`. I think I tried using `RuntimeExpressionRunner` for this first, but it made `ASan` unhappy. I'll give it another shot!

Reply:
Done. Seems to be working this time. :)